drivers/gles2/rasterizer_storage_gles2.cpp

```diff
@@ -5154,16 +5161,42 @@ void RasterizerStorageGLES2::render_target_set_external_texture(RID p_render_tar
 		t->alloc_height = rt->width;
 		t->alloc_width = rt->height;
 
-		// is there a point to setting the internal formats? we don't know them..
+		// Switch our texture on our frame buffer
+#if ANDROID_ENABLED
+		if (rt->msaa >= VS::VIEWPORT_MSAA_EXT_2X && rt->msaa <= VS::VIEWPORT_MSAA_EXT_4X) {
+			// This code only applies to the Oculus Go and Oculus Quest. Due to the the tiled nature
+			// of the GPU we can do a single render pass by rendering directly into our texture chains
+			// texture and apply MSAA as we render.
+
+			// On any other hardware these two modes are ignored and we do not have any MSAA,
+			// the normal MSAA modes need to be used to enable our two pass approach
```

Comment:
@BastiaanOlij There's no actual check that prevent this from actually being used on other Android devices, correct? By this I mean if an hardware manufacturer decides to follow Oculus's approach, there's nothing preventing a plugin for that hardware to leverage this logic.
So maybe we should leave the naming and description generic enough for Android devices.

Reply:
Nope, but it does only work on Android atm. I tried using it on desktop but it turns out there is a specific extention on Quest that allows us to do this.
I'm not sure what would be the best name, we shouldn't make it too broad or else people are going to try and use it on normal Android devices while this only works with the VR plugins.